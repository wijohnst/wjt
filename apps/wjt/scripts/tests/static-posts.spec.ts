import {
  getRawPosts,
  parseRawPosts,
  getFrontmatter,
  getRenderedPost,
} from '../source/static-posts';

import mockFs from 'mock-fs';
import { readFileSync } from 'fs';

const mockPostContent = [
  `---
    title: Post 1
    author: Some Author
    slug: post-1
    ---
    # Post 1

    This is the first post.
    `,
  `---
    title: Post 2
    author: Another Author
    slug: post-2
    ---
    # Post 2

    This is the second post.
    `,
];

describe('Static Posts', () => {
  beforeEach(() => {
    mockFs({
      'src/posts': {
        'post1.md': mockPostContent[0],
        'post1.html': `<div>Post 1</div>`,
        'post2.md': mockPostContent[1],
        'post2.html': `<div>Post 2</div>`,
      },
      'src/views/templates': {
        'head.pug': readFileSync('src/views/templates/head.pug', 'utf-8'),
      },
      'src/views/styles': {
        'min.css': readFileSync('src/views/styles/min.css', 'utf-8'),
      },
    });
  });

  describe('getRawPosts', () => {
    test('should be defined', () => {
      expect(getRawPosts).toBeDefined();
    });

    test('should return an array of posts', () => {
      const posts = getRawPosts();
      expect(posts).toHaveLength(2);
    });
  });

  describe('parseRawPosts', () => {
    test('should be defined', () => {
      expect(parseRawPosts).toBeDefined();
    });

    test('should return an array of parsed posts', () => {
      const parsedPosts = parseRawPosts();

      expect(parsedPosts[0].frontMatter).toEqual({
        title: 'Post 1',
        author: 'Some Author',
        slug: 'post-1',
      });

      const contentHunks = parsedPosts[0].content.split('\n');
      expect(contentHunks[0].trim()).toEqual('# Post 1');
      expect(contentHunks[1].trim()).toEqual('');
      expect(contentHunks[2].trim()).toEqual('This is the first post.');
    });
  });

  describe('getFrontmatter', () => {
    test('should be defined', () => {
      expect(getFrontmatter).toBeDefined();
    });

    test('should return the frontmatter of a post', () => {
      const frontmatter = getFrontmatter(mockPostContent[0]);
      expect(frontmatter).toEqual({
        title: 'Post 1',
        author: 'Some Author',
        slug: 'post-1',
      });
    });
  });

  describe('getRenderedPost', () => {
    test('should be defined', () => {
      expect(getRenderedPost).toBeDefined();
    });

    test('should return the rendered post', () => {
      mockFs.restore();

      const renderedPost = getRenderedPost({
        frontMatter: {
          title: 'Post 1',
          author: 'Some Author',
          slug: 'post-1',
        },
        content: '# Post 1\n\nThis is the first post.',
      });

      expect(renderedPost).toMatchSnapshot();
    });
  });
});
