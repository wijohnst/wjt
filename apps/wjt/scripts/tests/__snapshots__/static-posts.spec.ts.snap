// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Static Posts getRenderedPost should return the rendered post 1`] = `
"<head> <title>Post 1 </title><meta name="viewport" content="width=device-width, initial-scale=1"/><link rel="icon" href="https://wjt.sfo2.cdn.digitaloceanspaces.com/wjt_logo.ico"/></head><style>*,*:before,*:after{box-sizing:border-box}*{margin:0}body{line-height:1.5;-webkit-font-smoothing:antialiased}img,picture,video,canvas,svg{display:block;max-width:100%}input,button,textarea,select{font:inherit}p,h1,h2,h3,h4,h5,h6{overflow-wrap:break-word}p{text-wrap:pretty}h1,h2,h3,h4,h5,h6{text-wrap:balance}#root,#__next{isolation:isolate}html{padding:1rem;font-family:Roboto,sans-serif}.post{height:100%;border:solid thin black;padding:1rem}
</style><div class="post"><h1>Post 1</h1>
<p>This is the first post.</p>
</div>"
`;
